platform/src/main/java/org/hillview/sketches/FreqKListExact.java

```diff
@@ -0,0 +1,62 @@
+package org.hillview.sketches;
+
+import it.unimi.dsi.fastutil.objects.Object2IntMap;
+import it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap;
+import org.hillview.dataset.api.Pair;
+import org.hillview.table.Schema;
+import org.hillview.table.rows.RowSnapshot;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FreqKListExact extends FreqKList {
+
+    static Object2IntOpenHashMap<RowSnapshot> buildHashMap(List<RowSnapshot> rssList) {
+        Object2IntOpenHashMap<RowSnapshot> hMap = new Object2IntOpenHashMap<RowSnapshot>();
+        rssList.forEach(rss -> hMap.put(rss, 0));
+        return hMap;
+    }
+
+    public FreqKListExact(long totalRows, List<RowSnapshot> rssList, double epsilon) {
+        super(totalRows, epsilon, buildHashMap(rssList));
+    }
+
+    public FreqKListExact(List<RowSnapshot> rssList, double epsilon) {
+        super(0, epsilon, buildHashMap(rssList));
+    }
+
+    public FreqKListExact(long totalRows, double epsilon, Object2IntOpenHashMap<RowSnapshot> hMap) {
+        super(totalRows, epsilon, hMap);
+    }
+
+    /**
+     * Used to add two Lists that have counts for the same set of RowSnapShots. Behavior is not
+     * determined if it is called with two lists that have different sets of keys. Meant to be used
+     * by ExactFreqSketch.
+     * @param that The list to be added to the current one.
+     * @return Updated counts (existing counts are overwritten).
+     */
+    public FreqKListExact add(FreqKListExact that) {
```

Comment:
Can you please make this return a new object instead?
If objects are mutable very strange things happen when we use multiple threads.